### PR TITLE
Setup GitHub actions workflows

### PR DESCRIPTION
**Pull request by kmadisa**
Adding a workflow file to set up github actions to run the automated unit tests. The workflow uses a different docker image (hosted on https://artefact.skatelescope.org/) that has been created and maintained by the SKA System Team.

Also added a workflow, `publish_to_pypi.yaml`, to automate the process of publishing the Python package to [PyPI](https://pypi.org/project/tango-simlib/) when a tag is created. For more information on the action used to upload to PyPI, see https://github.com/marketplace/actions/pypi-publish.

For more information on github actions, see [link](https://docs.github.com/en/actions/learn-github-actions).

*Screenshots or code snippets (if appropriate):*
N/A

*Definition of Done Checklist*

- [x] ~Code meets our [python style guidelines]~(https://docs.google.com/document/d/1aZoIyR9tz5rCWr2qJKuMTmKp2IzHlFjrCFrpDDHFypM/edit?usp=sharing)?
- [x] ~Unit tested (coded, passed, included)?~
- [x] Requested at least 2 reviewers?
- [x] Commented code, particularly in hard-to-understand areas?
- [x] ~Made corresponding changes to the documentation (e.g. Python documentation, System Engineering Documentation, version description updates, README file, etc)?~

JIRA: [N/A](https://skaafrica.atlassian.net/browse/JIRA_ISSUE_NUMBER)

### CODE REVIEW

**Comment by kmadisa** [comment]
**NB: The failing tests will be fixed in PR https://github.com/ska-sa/mkat-tango/pull/205. The failures are consistent on both pipelines, so the github workflow is correct.**

File: .github/workflows/publish_to_pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: pypi release
+on:
+  create:
+    tags:
+      - 'v*.*.*'
+jobs:
+  build_and_publish_to_pypi:
```
Comment (mmphego) [inline .github/workflows/publish_to_pypi.yaml:7]: Do you not need to create a matrix, to set up both python2 and python3?

File: .github/workflows/unit_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Install flake8
+        run: sudo pip3 install flake8
+      - name: Get and tailor the sardana linting script
+        run: |
+          sudo apt-get install -y wget && \
+          wget -O lint_diff.sh https://raw.githubusercontent.com/sardana-org/sardana/develop/ci/flake8_diff.sh && \
+          sed -i 's+=sardana-org/sardana+=ska-sa/mkat-tango+g' lint_diff.sh && \
+          sed -i 's+develop+master+g' lint_diff.sh && \
+          sed -i 's+--show-source "$@"+--show-source "$@" --config=setup.cfg+g' lint_diff.sh && \
+          chmod +x lint_diff.sh
```
Comment (mmphego) [inline .github/workflows/unit_tests.yaml:20]: :+1: +1:

**Comment by mmphego** [comment]
Fix up the tests and we good to go!

**Comment by kmadisa** [comment]
> Fix up the tests and we good to go!

See https://github.com/ska-sa/mkat-tango/pull/204#issuecomment-908275364.

**Review by tockards — APPROVED** [review]
nice.

File: .github/workflows/publish_to_pypi.yaml (hunk 1 above)
Comment (kmadisa) [reply .github/workflows/publish_to_pypi.yaml:7]: We only catering for python 3.

**Pull request merged**